src/OpenTelemetry.ResourceDetectors.Container/ContainerResourceDetector.cs

```diff
@@ -132,45 +152,127 @@ private static string RemovePrefixAndSuffixIfNeeded(string input, int startIndex
         return input.Substring(startIndex, endIndex - startIndex);
     }
 
+    private static string? ExtractK8sContainerId()
+    {
+        try
+        {
+            var host = Environment.GetEnvironmentVariable(K8sServiceHostKey);
+            var port = Environment.GetEnvironmentVariable(K8sServicePortKey);
+            var @namespace = Environment.GetEnvironmentVariable(K8sNamespaceKey);
+            var hostname = Environment.GetEnvironmentVariable(K8sHostnameKey);
+            var containerName = Environment.GetEnvironmentVariable(K8sContainerNameKey);
+            var url = $"https://{host}:{port}/api/v1/namespaces/{@namespace}/pods/{hostname}";
+            var credentials = GetK8sCredentials(K8sCredentialPath);
+            if (string.IsNullOrEmpty(credentials))
+            {
+                return string.Empty;
+            }
+
+            using var httpClientHandler = ServerCertificateValidationHandler.Create(K8sCertificatePath, ContainerResourceEventSource.Log);
+            var response = ResourceDetectorUtils.SendOutRequest(url, "GET", new KeyValuePair<string, string>("Authorization", credentials), httpClientHandler).GetAwaiter().GetResult();
```

Comment:
I'm very nervous with using sync-over-async and potential dead-lock if some `ConfigureAwait(false)` is not done. Can we use [Send ](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.send) instead? Or would it create problems with default client handler or if `AllowSynchronousIO = false` for ASP.NET Core.

Otherwise, it may be safe to use:
```
    private static readonly TaskFactory _taskFactory = new
        TaskFactory(CancellationToken.None,
            TaskCreationOptions.None,
            TaskContinuationOptions.None,
            TaskScheduler.Default);

    public static T RunSync<T>(Func<Task<T>> task)
        => _taskFactory
            .StartNew(task)
            .Unwrap()
            .GetAwaiter()
            .GetResult();
```

Reply:
We have the same issue in all AWS resource detectors as they are using the same `ResourceDetectorUtils.SendOutRequest`

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/c81572ee8d5dafb20c91486f967c0935c5290adf/src/OpenTelemetry.ResourceDetectors.AWS/AWSEC2ResourceDetector.cs#L95

Is it ok for you if I fix this issue for all detectors in a follow-up PR?